x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/ConnectorIndexServiceTests.java

```diff
@@ -291,6 +293,44 @@ public void testUpdateConnectorFiltering_updateDraft() throws Exception {
         );
     }
 
+    public void testUpdateConnectorFiltering_updateDraftWithEmptyRules() throws Exception {
```

Comment:
can we/ should we add a unit tests that handles another edge case:
- update draft with just a `[default rule]` to make sure we can handle it